src/components/questionnaire/questionnaire-interaction-screen.tsx

```diff
@@ -0,0 +1,160 @@
+import { KeyboardReturn } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Checkbox,
+  Divider,
+  FormControlLabel,
+  Typography
+} from "@mui/material";
+import { useState } from "react";
+import type { AnswerOption, Question, Questionnaire } from "src/generated/homeLambdasClient";
+import strings from "src/localization/strings";
+
+interface QuestionnaireInteractionScreenProps {
+  questionnaire: Questionnaire | null;
+  mode: "view" | "edit" | "preview";
+  onBack: () => void;
+}
+
+/**
+ *  Screen for the user to interact with the questionnaire; fill, edit and preview
+ */
+const QuestionnaireInteractionScreen: React.FC<QuestionnaireInteractionScreenProps> = ({
+  questionnaire,
+  mode,
+  onBack
+}) => {
+  if (!questionnaire) {
+    return <Typography variant="h4">Loading...</Typography>;
+  }
```

Comment:
Use a loading spinner for loading state, or perhaps a string saying "currently no questionnaire". Loading implies the user should expect to wait for something to change